From 9db672f1774b501e87a05292f9fd97dfbba96a22 Mon Sep 17 00:00:00 2001
From: Ban Feng <kcfeng0@nuvoton.com>
Date: Tue, 6 Jun 2023 11:33:34 +0800
Subject: [PATCH] Create Associations From Defined Property

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Ban Feng <kcfeng0@nuvoton.com>
---
 src/entity_manager/entity_manager.cpp | 30 +++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/src/entity_manager/entity_manager.cpp b/src/entity_manager/entity_manager.cpp
index 5fcee7576..37fa879dc 100644
--- a/src/entity_manager/entity_manager.cpp
+++ b/src/entity_manager/entity_manager.cpp
@@ -34,6 +34,7 @@
 #include <functional>
 #include <map>
 #include <regex>
+#include <iostream>
 constexpr const char* tempConfigDir = "/tmp/configuration/";
 constexpr const char* lastConfiguration = "/tmp/configuration/last.json";
 
@@ -194,6 +195,35 @@ void EntityManager::postBoardToDBus(
         }
     }
 
+    auto findAssociations = boardValues.find("Associations");
+    if (findAssociations != boardValues.end() &&
+        findAssociations->type() == nlohmann::json::value_t::array)
+    {
+        std::cerr << "create Associations to " << boardNameOrig << "\n";
+        std::vector<Association> associations;
+        for (auto& association : *findAssociations)
+        {
+            auto findForward = association.find("Forward");
+            auto findReverse = association.find("Reverse");
+            auto findPath = association.find("Path");
+            if (findForward != association.end() &&
+                findReverse != association.end() &&
+                findPath != association.end())
+            {
+                associations.emplace_back(
+                    findForward->get<std::string>(),
+                    findReverse->get<std::string>(),
+                    findPath->get<std::string>());
+            }
+        }
+        std::shared_ptr<sdbusplus::asio::dbus_interface> ifacePtr =
+            dbus_interface.createInterface(
+                boardPath, "xyz.openbmc_project.Association.Definitions",
+                boardNameOrig);
+        ifacePtr->register_property("Associations", associations);
+        ifacePtr->initialize();
+    }
+
     nlohmann::json::iterator exposes = boardValues.find("Exposes");
     if (exposes == boardValues.end())
     {
-- 
2.43.0

