
diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
index 7247e2eb193e..b2d2363658df 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
@@ -127,6 +127,11 @@ tip_reserved: tip@0x0 {
 			reg = <0x0 0x0 0x0 0x6200000>;
 		};
 
+		espi_flash_memory: espi_flash@0x10000000 {
+			reg = <0x0 0x10000000 0x0 0x4000000>;
+			no-map;
+		};
+
 		/*linux,cma {
 			compatible = "shared-dma-pool";
 			reusable;
@@ -210,6 +215,11 @@ &gmac3 {
 	status = "okay";
 };
 
+&espi_flash {
+	memory-region = <&espi_flash_memory>;
+	status = "okay";
+};
+
 &tmps {
 	status = "okay";
 };
