From fbbd1e5098b2411e3eb6b7eb08689b1bc30f47d2 Mon Sep 17 00:00:00 2001
From: Allen Kang <jhkang@nuvoton.com>
Date: Tue, 21 Feb 2023 17:07:48 +0800
Subject: [PATCH] powerOn always return true

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Allen Kang <jhkang@nuvoton.com>
---
 power-supply/power_supply.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/power-supply/power_supply.cpp b/power-supply/power_supply.cpp
index f2c2e87..c7a5bb6 100644
--- a/power-supply/power_supply.cpp
+++ b/power-supply/power_supply.cpp
@@ -246,7 +246,7 @@ void PowerSupply::powerStateChanged(sdbusplus::message_t& msg)
 
 void PowerSupply::updatePowerState()
 {
-    powerOn = util::isPoweredOn(bus);
+    powerOn = true;
 }
 
 void PowerSupply::checkInputFault(const uint16_t statusWord)
