From 8848fa7923ed02eed21183d1ef0b7c923dd14a56 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Mon, 2 Jun 2025 09:34:00 +0800
Subject: [PATCH] Enable DVO HSYNC, DDC i2c port

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
Upstream-Status: Inappropriate [oe-specific]
---
 board/nuvoton/arbel_evb/arbel_evb.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/board/nuvoton/arbel_evb/arbel_evb.c b/board/nuvoton/arbel_evb/arbel_evb.c
index 954aaf3f36..f7aa34ad6a 100644
--- a/board/nuvoton/arbel_evb/arbel_evb.c
+++ b/board/nuvoton/arbel_evb/arbel_evb.c
@@ -24,6 +24,12 @@ DECLARE_GLOBAL_DATA_PTR;
 
 int board_init(void)
 {
+	/* Set INTCR [27]=1b for HSYNC enable */
+	setbits_le32(0xF080003C, BIT(27));
+	/* Set MFSEL4 [30:29]=10b, SMB11 for Graphics DDC port */
+	clrbits_le32(0xF080026C, BIT(29));
+	setbits_le32(0xF080026C, BIT(30));
+
 	return 0;
 }
 
-- 
2.43.0

