From 13c7c17fff96746390a41de2c25790a992fb76fe Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 14 Feb 2025 15:32:45 +0800
Subject: [PATCH] driver: mctp usb: set tx_queue_len

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 drivers/net/mctp/mctp-usb.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/mctp/mctp-usb.c b/drivers/net/mctp/mctp-usb.c
index 1dd406ab0ebb..52bf7bd02609 100644
--- a/drivers/net/mctp/mctp-usb.c
+++ b/drivers/net/mctp/mctp-usb.c
@@ -11,6 +11,7 @@
 #include <linux/usb/mctp-usb.h>
 
 #include <net/mctp.h>
+#include <net/pkt_sched.h>
 
 #include <uapi/linux/if_arp.h>
 
@@ -277,6 +278,7 @@ static void mctp_usb_netdev_setup(struct net_device *dev)
 	dev->max_mtu = MCTP_USB_MTU_MAX;
 
 	dev->hard_header_len = sizeof(struct mctp_usb_hdr);
+	dev->tx_queue_len = DEFAULT_TX_QUEUE_LEN;
 	dev->addr_len = 0;
 	dev->flags = IFF_NOARP;
 	dev->netdev_ops = &mctp_usb_netdev_ops;
-- 
2.34.1

